cmd/config/docs/api-conventions/functions-spec.md

```diff
@@ -18,168 +23,372 @@ interpreted as described in [RFC 2119][2].
 ```
 
-In addition, a function MUST accept as input a List of kind `ResourceList` where the
-`functionConfig` field, if present, will contain the invocation-specific configuration passed to the function
-by the orchestrator.
-Functions MAY consider this field optional so that they can be triggered in an ad-hoc fashion.
+#### Examples
 
-An example using `config.kubernetes.io/v1beta1/ResourceList` as input:
+The following is an example input, where the custom resource of kind
+`FulfillmentCenter` is provided as `functionConfig`. The function will operate
+on one resource of kind `Service`.
 
 ```yaml
-apiVersion: config.kubernetes.io/v1beta1
+apiVersion: config.kubernetes.io/v1
 kind: ResourceList
 functionConfig:
   apiVersion: foo-corp.com/v1
   kind: FulfillmentCenter
   metadata:
     name: staging
-    metadata:
-      annotations:
-        config.kubernetes.io/function: |
-          container:
-            image: gcr.io/example/foo:v1.0.0
   spec:
     address: "100 Main St."
 items:
-  - apiVersion: rbac.authorization.k8s.io/v1
-    kind: ClusterRole
+  - apiVersion: v1
+    kind: Service
     metadata:
-      name: namespace-reader
-    rules:
-      - resources:
-          - namespaces
-        apiGroups:
-          - ""
-        verbs:
-          - get
-          - watch
-          - list
+      name: wordpress
+      labels:
+        app: wordpress
+      annotations:
+        config.kubernetes.io/index: "0"
+        config.kubernetes.io/path: "service.yaml"
+    spec: # Example comment
+      type: LoadBalancer
+      selector:
+        app: wordpress
+        tier: frontend
+      ports:
+        - protocol: TCP
+          port: 80
 ```
 
-Here `FulfillmentCenter` kind with name `staging` is passed as the invocation-specific configuration
-to the function.
+The following is an example output containing one result representing a
+validation error:
 
-### Output Type
-
-A function’s output MUST be the same as the input specification above
--- i.e. `ResourceList` or `List`.
-This is necessary to enable chaining two or more functions together in a pipeline.
-The serialization format of the output SHOULD match that of its input on each invocation
--- e.g. if the input was a `ResourceList`, the output should also be a `ResourceList`.
+```yaml
+apiVersion: config.kubernetes.io/v1
+kind: ResourceList
+items:
+  - apiVersion: v1
+    kind: Service
+    metadata:
+      name: wordpress
+      labels:
+        app: wordpress
+      annotations:
+        config.kubernetes.io/index: "0"
+        config.kubernetes.io/path: "service.yaml"
+    spec: # Example comment
+      type: LoadBalancer
+      selector:
+        app: wordpress
+        tier: frontend
+      ports:
+        - protocol: TCP
+          port: 80
+results:
+  - message: "Invalid type. Expected: integer, given: string"
+    severity: error
+    resourceRef:
+      apiVersion: v1
+      kind: Service
+      name: wordpress
+    field:
+      path: spec.ports.0.port
```

Comment:
Should it be `spec.ports[0].port`? I'm not sure if what we have here is valid jsonpath.

Reply:
I think all following paths are valid here:
- `spec.ports[0].port`
- `spec.ports.[0].port`
- `spec.ports.0.port`